lib/python/treadmill_aws/cli/admin/aws/instance.py

```diff
@@ -116,26 +117,68 @@ def create(
         image_id = aws_cli.admin.image_id(
             ec2_conn, sts_conn, image, image_account)
         secgroup_id = aws_cli.admin.secgroup_id(ec2_conn, secgroup)
-        subnet_id = aws_cli.admin.subnet_id(ec2_conn, subnet)
+
+        total_subnets = []
 
         if not key:
             key = metadata.instance_keys()[0]
 
-        hostnames = hostmanager.create_host(
-            ipa_client=ipa_client,
-            ec2_conn=ec2_conn,
-            image_id=image_id,
-            count=count,
-            disk=disk_size,
-            domain=ipa_domain,
-            key=key,
-            secgroup_ids=secgroup_id,
-            instance_type=size,
-            subnet_id=subnet_id,
-            role=role,
+        def run_ec2(count, subnet_id):
```

Comment:
this entire logic should be moved to hostmanager. argument to hostmanages is list of subnet ids, and it will select in hostmanager.create_host.

it should be separate methods for each:
 - get # of hosts in subnet
 - is_space_availble

placements = {}
for subnet in subnets:
   available, total = get_availability(subnet)
   placements[subnet] = (availability, total)

for subnet, availability in placemens:
  - calculate best placement - % of allocated, min servers, whatever.

return best_subnet.